Refactored code to MVP

mrsujan06 opened the pull request:
Followed MVP design pattern. Added BaseActivity, BasePresenter, MvpView , changed
method names for better understanding of what method does.

mrsujan06 committed:
Refactored code to MVP

Followed MVP design pattern. Added BaseActivity, BasePresenter, MvpView , changed
method names for better understanding of what method does.

uchitsingh commented on .idea/misc.xml:
```diff
@@ -29,7 +29,7 @@
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8 (1)" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8 (1)" project-jdk-type="JavaSDK">
```
not sure what  this change affects, guess we can revert this before merging to master. usually these files are set to be git ignored, and specific to developers local setting.

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/base/BaseActivity.java:
```diff
@@ -0,0 +1,80 @@
+public class BaseActivity extends AppCompatActivity implements MvpView {
+
+    private ProgressDialog mProgressDialog;
+
```
do not leave double line

uchitsingh replied:
do not leave double line, seperate only by 1 line

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/base/BaseActivity.java:
```diff
@@ -0,0 +1,80 @@
+                checkSelfPermission(permission) == PackageManager.PERMISSION_GRANTED;
+    }
+
+
```
do not leave double lines, seperate by 1

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/base/BaseActivity.java:
```diff
@@ -0,0 +1,80 @@
+        } else {
+            showSnackBar(getString(R.string.some_error));
+        }
+
```
delete extra line , 
Also, try to get in habbit of ctrl + alt+l  to autoformat.,

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/base/BasePresenter.java:
```diff
@@ -0,0 +1,28 @@
+        return mApiNepalServiceInteractor;
+    }
+
+
```
remove extra line, seperate only by 1

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/base/BasePresenter.java:
```diff
@@ -0,0 +1,28 @@
+
+public class BasePresenter<V extends MvpView> {
+
+    private V mMvpView;
```
Usually,  the ordering is 

Public static final 
protected static final
private static final 
public String
protected String
private String

uchitsingh replied:
protected ApiNepalServiceInteractor mApiNepalServiceInteractor;
 private V mMvpView;

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/camping/CampingActivity.java:
```diff
@@ -31,21 +33,36 @@ protected void onCreate(Bundle savedInstanceState) {
 
         String place_name = (String) getIntent().getExtras().getSerializable("cityname");
-        mNepalPresenter.campingSpotCall("camping in " + place_name.toUpperCase(), "AIzaSyBT2bl_XWXG7-fsWtCNyGrTD8wFxaBxbTc");
+        mCampingPresenter.listAllCampingSpots("Camping+in+" + place_name.toUpperCase(), "AIzaSyBT2bl_XWXG7-fsWtCNyGrTD8wFxaBxbTc");
```
???
handling logic   in activity?

uchitsingh replied:
pass only place_name in listAllCampingSpots(place_name, apiKey)
make apiKey private static final apikeyCampingSpot? as a localvariable, and use it 

IN presenter, u can ammend logic by getting the query, and adding in whatever needed, there
query =  "Camping+in+" + query.toUpperCase()

uchitsingh replied:
think u have apikey define as static in Apilist

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/camping/CampingAdapter.java:
```diff
@@ -71,7 +70,7 @@ public CampingAdapterViewHolder(View itemView) {
         @Override
         public void onClick(View v) {
-            Intent intent = new Intent(v.getContext(), CampingSpot.class);
+            Intent intent = new Intent(v.getContext(), CampingActivity.class);
```
Make a static context in App Class, and get the context from there??

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/camping/CampingPresenter.java:
```diff
@@ -0,0 +1,59 @@
+public class CampingPresenter extends BasePresenter<CampingView> {
+
+    private static final String TAG = "ERROR MESSAGE";
+
```
remove extraline

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/camping/CampingPresenter.java:
```diff
@@ -0,0 +1,59 @@
+
+    @SuppressLint("CheckResult")
+    public void listAllCampingSpots(String query, String key) {
+
```
handle the query logic here //

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/camping/CampingPresenter.java:
```diff
@@ -0,0 +1,59 @@
+
+    }
+
+
```
delete xtra line

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/cashMachine/CashMachineActivity.java:
```diff
@@ -32,30 +31,33 @@ protected void onCreate(Bundle savedInstanceState) {
-        mNepalPresenter.cashMachineCall("ATM in " + place_name.toUpperCase(), "AIzaSyBT2bl_XWXG7-fsWtCNyGrTD8wFxaBxbTc");
-
-
+        mCashMachinePresenter.listAllCashMachine("ATM+in+" + place_name.toUpperCase(), API_KEY);
```
handle logic in presenter

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/dagger/AppComponent.java:
```diff
@@ -1,19 +1,19 @@
+    void inject(CampingActivity campingActivity);
+    void inject(RestaurantActivity restaurantActivity);
+    void inject(CashMachineActivity cashMachineActivity);
     void inject(Mountain mountain);
```
?MountainActivity?

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/home/VisitNepalActivity.java:
```diff
@@ -0,0 +1,71 @@
+
+public class VisitNepalActivity extends BaseActivity implements VisitNepalView {
+
+
```
Remove extra line

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/model/Interactor/ApiServiceInteractorImp.java:
```diff
@@ -36,13 +36,18 @@ public ApiServiceInteractorImp() {
+        return mService.getMountains();
+    }
+
 
```
remove extra line

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/mountain/MountainPresenter.java:
```diff
@@ -0,0 +1,56 @@
+    public MountainPresenter(ApiNepalServiceInteractor apiNepalServiceInteractor) {
+        this.mApiNepalServiceInteractor = apiNepalServiceInteractor;
+    }
+
```
remove extra line

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/placeOptions/PlacesOptions.java:
```diff
@@ -7,36 +7,35 @@
 import butterknife.OnClick;
 
 
 public class PlacesOptions extends AppCompatActivity {
```
PlacesOptionActivity?

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/restaurant/RestaurantActivity.java:
```diff
@@ -0,0 +1,75 @@
+    RestaurantPresenter mRestaurantPresenter;
+    RestaurantAdapter mAdapter;
+
+
```
Remove extra line

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/restaurant/RestaurantActivity.java:
```diff
@@ -0,0 +1,75 @@
+        String place_name = (String) getIntent().getExtras().getSerializable("cityname");
+        mRestaurantPresenter.listAllRestaurants("Restaurants+in+" + place_name, API_KEY);
+
+
```
Remove extra line

uchitsingh commented on app/src/main/java/com/example/arvin/nepaltouristguide/restaurant/RestaurantPresenter.java:
```diff
@@ -0,0 +1,56 @@
+        this.mApiNepalServiceInteractor = apiNepalServiceInteractor;
+    }
+
+
```
Remove extra line

mrsujan06 committed:
Fix/Deleted double lines

Deleted double lines , Deleted unused imports and changed language level
to JDK_1_8 on misc.xml file

mrsujan06 committed:
Deleted double lines , edited some classes

Deleted double lines , Deleted unused imports and changed language level
to JDK_1_8 on misc.xml file , edited App class

mrsujan06 committed:
Merge branch 'Refactoringcode' of https://github.com/mrsujan06/Visit-Nepal into Refactoringcode

uchitsingh commented on app/src/main/AndroidManifest.xml:
```diff
@@ -6,28 +6,27 @@
+        <activity android:name=".camping.CampingActivity" />
+        <activity android:name=".restaurant.RestaurantActivity" />
+        <activity android:name=".cashMachine.CashMachineActivity" />
+        <activity android:name=".mountain.Mountain"></activity>
```
MountainActivity?
